File: dujiaoka-fastapi/backend/app/schemas/order.py
"""
订单相关的Pydantic模型
"""
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, Field

from .product import Product
from .user import User


class OrderBase(BaseModel):
    """订单基础模型"""
    product_id: int
    quantity: int = Field(1, gt=0)
    payment_method: str = Field(..., pattern="^(balance|alipay|wechat)$")
    user_note: Optional[str] = None


class OrderCreate(OrderBase):
    """订单创建模型"""
    pass


class OrderUpdate(BaseModel):
    """订单更新模型"""
    status: Optional[str] = None
    admin_note: Optional[str] = None


class Order(OrderBase):
    """订单模型"""
    id: int
    order_number: str
    user_id: int
    product_name: str
    product_price: float
    total_amount: float
    status: str
    delivery_content: Optional[str]
    delivered_at: Optional[datetime]
    admin_note: Optional[str]
    created_at: datetime
    updated_at: datetime
    paid_at: Optional[datetime]

    # 关联对象
    user: Optional[User]
    product: Optional[Product]

    class Config:
        from_attributes = True


class OrderList(BaseModel):
    """订单列表响应"""
    items: List[Order]
    total: int
    page: int
    size: int


class CartItem(BaseModel):
    """购物车项"""
    product_id: int
    quantity: int = Field(1, gt=0)


class OrderSummary(BaseModel):
    """订单摘要"""
    items: List[CartItem]
    total_amount: float

File: dujiaoka-fastapi/frontend/src/types/order.ts
// 订单相关类型定义

export interface Order {
  id: number
  order_number: string
  user_id: number
  product_id: number
  product_name: string
  product_price: number
  quantity: number
  total_amount: number
  payment_method: string
  status: string
  delivery_content?: string
  delivered_at?: string
  user_note?: string
  admin_note?: string
  created_at: string
  updated_at: string
  paid_at?: string
  user?: {
    id: number
    username: string
    email: string
  }
  product?: {
    id: number
    name: string
    image_url?: string
  }
}

export interface OrderCreate {
  product_id: number
  quantity: number
  payment_method: string
  user_note?: string
}

export interface OrderUpdate {
  status?: string
  admin_note?: string
}

export interface OrderList {
  items: Order[]
  total: number
  page: number
  size: number
}

export interface CartItem {
  product_id: number
  quantity: number
  price?: number
  name?: string
  image_url?: string
}

export interface OrderSummary {
  items: CartItem[]
  total_amount: number
}
